core/src/main/scala/chisel3/FileDescriptor.scala

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import chisel3._
+import chisel3.internal.firrtl.ir.Printf
+import chisel3.internal.Builder
+import chisel3.experimental.SourceInfo
+
+/** A file or I/O device to print to in simulation
+  * 
+  * {{{
+  * val fd = FileDescriptor("logfile.log")
+  * fd.printf(cf"in = $in%0d\n")
+  * }}}
+  */
+sealed trait FileDescriptor extends FileDescriptorIntf {
+
+  /** Prints a message in simulation
+    *
+    * Prints a message every cycle. If defined within the scope of a [[when]] block, the message
+    * will only be printed on cycles that the when condition is true.
+    *
+    * Does not fire when in reset (defined as the encapsulating Module's reset). If your definition
+    * of reset is not the encapsulating Module's reset, you will need to gate this externally.
+    *
+    * May be called outside of a Module (like defined in a function), uses the current default clock
+    * and reset. These can be overriden with [[withClockAndReset]].
+    *
+    * @see [[Printable]] documentation
+    * @param pable [[Printable]] to print
+    */
+  def printf(pable: Printable)(implicit sourceInfo: SourceInfo): chisel3.printf.Printf = {
+    this.printfWithReset(pable)(sourceInfo)
+  }
+
+  // The actually underlying file descriptor, None means just use printf
+  protected def fileDescriptor: Option[String]
```

Comment:
To move towards unifying this with stdout/stderr, introducing a new enum of: `File(name: String) | StdErr` would be a way to start. This would also avoid the question of "what is the default" when this is `None`. This _should_ unify fprintf with printf from the Chisel side and enable moving in the direction of `LogFile` or something like that?